src/components/TagPicker/index.tsx

```diff
@@ -67,13 +83,29 @@ function TagPicker({selectedTag, tagListName, policyID, tagListIndex, shouldShow
     }, [selectedTag]);
 
     const enabledTags: PolicyTags | Array<PolicyTag | SelectedTagOption> = useMemo(() => {
-        if (!shouldShowDisabledAndSelectedOption) {
+        if (!shouldShowDisabledAndSelectedOption && !hasDependentTags) {
             return policyTagList.tags;
         }
+
+        if (!shouldShowDisabledAndSelectedOption && hasDependentTags) {
+            // Truncate transactionTag to the current level (e.g., "California:North")
+            const parentTag = transactionTag?.split(':').slice(0, tagListIndex).join(':');
```

Comment:
We are not truncating the current level tags correctly which caused an issue https://github.com/Expensify/App/issues/68777